Rubberduck.Parsing/Symbols/Declaration.cs

```diff
@@ -36,7 +36,7 @@ public class Declaration : IEquatable<Declaration>
                 "ANY"
         };
 
-        public static readonly IDictionary<string, string> TYPEHINT_TO_TYPENAME = new Dictionary<string, string>
```

Comment:
That dictionary should have been exposed as an `IReadOnlyDictionary<string,string>`, these mappings aren't meant to be tampered with. I'm still not convinced about exposing a public field here, when an instance property could do the job and simply expose a `HintedTypeName` and abstract away the data, the implementation detail that is this dictionary.